### PR TITLE
Add todo functionality

### LINKED ISSUE #11: Creating add todo functionality
generate todos using form table

### PR DESCRIPTION
**Pull request by dgupta0**
fixes #11

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -227,54 +220,64 @@ function App() {
...
-            newArr[i] = { ...todo }
-          }
-        }
-        return newArr
-      })
-    }
-    else {
-      todo.tags = todo.tags ? todo.tags.split(",") : []
+    if (todo.title && todo.description && todo.status) {
+      todo.tags = todo.tags.length ? todo.tags.split(",") : []
```
Comment (Dorfieeee) [inline src/App.jsx:244]:
```suggestion
todo.tags = todo.tags.length ? todo.tags.split(",").map(tag => tag.trim()).filter(tag => tag.length !== 0) : []
```

File: src/App.jsx [hunk 2]
```diff
@@ -227,54 +220,64 @@ function App() {
...
+        setData(data => {
+          let newArr = [...data]
+          for (let i = 0; i < newArr.length; i++) {
+            if (newArr[i].id === todo.id) {
+              newArr[i] = { ...todo }
+            }
+          }
+          return newArr
+        })
+      }
```
Comment (Dorfieeee) [inline src/App.jsx:261]: I would also move this to a separate function. `onFinishCreate`, `onFinishEdit`, ...

File: src/App.jsx (hunk 1 above)
Comment (dgupta0) [reply src/App.jsx:244]: Why have we made this function so complex and why is it better?

File: src/App.jsx (hunk 2 above)
Comment (dgupta0) [reply src/App.jsx:261]: For that I will have to make a different button for edit. Right now the `Done` btn is handling the cases for both edit and add todo.

**Pull request merged**